### PR TITLE
Add larger scheduler_perf test cases

### PR DESCRIPTION
**Pull request by macsko**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

We want to setup scheduling throughput thresholds for scheduler_perf test cases. Before doing that, we need to make the cases running longer to get more accurate results for each case. This PR increases number or `measurePods` for benchmarks that are too noisy according to [perf-dash](https://perf-dash.k8s.io/#/?Metric=SchedulingThroughput&Name=SchedulingBasic%2F5000Nodes%2Fnamespace-2&extension_point=not%20applicable&plugin=not%20applicable&result=not%20applicable) charts.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sanposhiho** [comment]
Memo: we may have to increase the timeout / interval after adding those new test cases. Let's confirm after merging the PR.

https://github.com/kubernetes/test-infra/blob/0a0ecca37884da4c760bcb193c4fb85c3977b2e6/config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml#L721

**Comment by macsko** [comment]
@pohly @sanposhiho It seems etcd is working differently than expected. 

After adding larger test cases, the benchmark returns `etcdserver: mvcc: database space exceeded` error at some point. It turns out, all test cases share the same etcd that is started by `make test-integration` script [here](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/hack/make-rules/test-integration.sh#L73). Then, calling `startEtcd()` during benchmark setup ([ref](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/test/integration/scheduler_perf/scheduler_perf.go#L781)), doesn't do anything, because it verifies that the etcd is already running [here](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/test/integration/framework/etcd.go#L71). The test cases aren't inferring each other supposedly because different storage prefixes passed to test api server ([ref](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/test/integration/scheduler_perf/util.go#L100)) created randomly [here](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/test/integration/framework/controlplane_utils.go#L97).

We don't cleanup the etcd for given prefix after each test case, what result in storing the items from previous cases, what eventually lead to `database space exceeded` error.

**Comment by sanposhiho** [comment]
So, can't we increase the capacity of etcd somehow?

**Comment by macsko** [comment]
> So, can't we increase the capacity of etcd somehow?

I think it could be possible. Etcd is created [here](https://github.com/kubernetes/kubernetes/blob/b95f9c32d65638b63dee7fc887ff9ab2ba409c58/hack/lib/etcd.sh#L87) without storage size limit specified. According to etcd [docs](https://etcd.io/docs/v3.3/dev-guide/limit/#:~:text=The%20default%20storage%20size%20limit,quota%2Dbackend%2Dbytes%20flag.) it's 2GB by default, but can be increased.

**Comment by pohly** [comment]
> make test-integration

That wasn't how I tested: I ran `go test`, so there was no shared etcd. The conclusion still stands: restarting etcd and not cleaning up is faster.

> So, can't we increase the capacity of etcd somehow?

If we share the same etcd with other integration tests, then we should clean up.

Can we parameterize `startEtcd` so that it doesn't reuse the same etcd as the other tests? A crude hack would be to unset whatever env variables cause it to reuse the running one.

**Comment by macsko** [comment]
/cc @pohly @sanposhiho 

> Can we parameterize `startEtcd` so that it doesn't reuse the same etcd as the other tests

Did that and it seems It's working correctly ([test1](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126337/pull-kubernetes-scheduler-perf/1816499649951830016), [test2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126337/pull-kubernetes-scheduler-perf/1816479564667293696)).

File: test/integration/scheduler_perf/config/performance-config.yaml [hunk 1]
```diff
@@ -35,6 +35,12 @@
       initNodes: 5000
```
Comment (sanposhiho) [inline test/integration/scheduler_perf/config/performance-config.yaml:35]: Should we just remove `performance` tag from the existing medium ones (`5000Nodes`)?
If it's too noisy like you said, I don't think it's worth keeping in perf-dash.

Comment (macsko) [reply test/integration/scheduler_perf/config/performance-config.yaml:35]: Good idea, removed

**Review by sanposhiho — COMMENTED** [review]
I see test cases have different `measurePods` from 2000 to 10000; should we make all of them `10000` like `SchedulingBasic`? (I'm here assuming the more measure Pods it has, the more stable the result would be) 
Or would it take too long time to complete?

**Comment by macsko** [comment]
> I see test cases have different `measurePods` from 2000 to 10000; should we make all of them `10000` like `SchedulingBasic`? (I'm here assuming the more measure Pods it has, the more stable the result would be) Or would it take too long time to complete?

Unfortunately some of the test cases have much smaller throughputs, so the measurements of 10000 pods would take too long to complete. The tests will last sufficiently long to be stable.

**Review by sanposhiho — COMMENTED** [review]
/lgtm

File: test/integration/scheduler_perf/config/performance-config.yaml [hunk 2]
```diff
@@ -30,11 +30,17 @@
       initPods: 500
       measurePods: 1000
   - name: 5000Nodes
-    labels: [performance, fast]
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/config/performance-config.yaml:33]: in which jobs will this test case run, after the change?

File: test/integration/scheduler_perf/scheduler_perf.go [hunk 3]
```diff
@@ -778,7 +778,7 @@ func setupTestCase(t testing.TB, tc *testCase, output io.Writer, outOfTreePlugin
 
 	// Now that we are ready to run, start
 	// etcd.
```
Comment (alculquicondor) [inline test/integration/scheduler_perf/scheduler_perf.go:780]:
```suggestion
	// a brand new etcd.
```

File: test/integration/scheduler_perf/config/performance-config.yaml (hunk 2 above)
Comment (macsko) [reply test/integration/scheduler_perf/config/performance-config.yaml:33]: In no test, but could be run additionally by setting the test selector to `fast`

**Comment by alculquicondor** [comment]
/approve

Could you add some notes about the available labels in https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf#benchmark-tests?

**Comment by alculquicondor** [comment]
Nvm, that's in the other PR

**Pull request merged**